libraries/botbuilder/src/botFrameworkAdapter.ts

```diff
@@ -1195,8 +1196,17 @@ export class BotFrameworkAdapter
             if (request.deliveryMode === DeliveryModes.ExpectReplies) {
                 // Handle "expectReplies" scenarios where all the activities have been buffered and sent back at once
                 // in an invoke response.
-                const expectedReplies: ExpectedReplies = { activities: context.bufferedReplyActivities as Activity[] };
-                body = expectedReplies;
+                let activities = context.bufferedReplyActivities as Activity[];
+                
+                // If the channel is not the emulator, do not send trace activities.
+                // Fixes: https://github.com/microsoft/botbuilder-js/issues/2732
+                if (request.channelId !== Channels.Emulator) {
+                    activities = context.bufferedReplyActivities.reduce(
```

Comment:
Seems like `filter` would do this in a more concise way: `activities = activities.filter(a => a.type === ActivityTypes.Trace)`

Reply:
Good call, I don't know why I led with `.reduce()`...

Reply:
Reduce is a good one too 😄